plugins/dynamodb-positions-storage/src/main/java/com/github/bsideup/liiklus/dynamodb/DynamoDBPositionsStorage.java

```diff
@@ -51,57 +43,27 @@
 
     public static final String RANGE_KEY_FIELD = "groupId";
 
-    AmazonDynamoDBAsync dynamoDB;
+    DynamoDbAsyncClient dynamoDB;
 
     String tableName;
 
     @Override
     public Publisher<Positions> findAll() {
-        var request = new ScanRequest(tableName);
-
-        AtomicBoolean done = new AtomicBoolean(false);
-
-        return Mono
-                .<ScanResult>create(sink -> dynamoDB.scanAsync(request, new DefaultAsyncHandler<>(sink)))
-                .<List<Map<String, AttributeValue>>>handle(((scanResult, sink) -> {
-                    try {
-                        if (scanResult.getCount() <= 0) {
-                            done.set(true);
-                            sink.complete();
-                        } else {
-                            if (scanResult.getLastEvaluatedKey() == null || scanResult.getLastEvaluatedKey().isEmpty()) {
-                                done.set(true);
-                            }
-                            sink.next(scanResult.getItems());
-                        }
-                    } catch (Exception e) {
-                        sink.error(e);
-                    }
-                }))
-                .repeat(() -> !done.get())
-                .flatMapIterable(it -> it)
+        return Flux.defer(() -> Flux.from(dynamoDB.scanPaginator(req -> req.tableName(tableName))))
```

Comment:
Why `defer` here? Doesn't SDK 2 respects Publisher semantics and return a cold publisher?

Reply:
>When the operation is called, an instance of this class is returned. At this point, no service calls are made yet

ok, will fix that